scheduler/src/cook/kubernetes/api.clj

```diff
@@ -342,6 +349,49 @@
                    (not (contains? disallowed-var-names (.getName var)))))
          (into []))))
 
+(defn- add-as-decimals
+  "Takes two doubles and adds them as decimals to avoid floating point error. Kubernetes will not be able to launch a
+   pod if the required cpu has too much precision. For example, adding 0.1 and 0.02 as doubles results in 0.12000000000000001"
+  [a b]
+  (-> a
+    BigDecimal/valueOf
+    (.add (BigDecimal/valueOf b))
+    .doubleValue))
+
+(defn adjust-job-resources-fn
+  "Given the required resources for a job, add to them the resources for any sidecars that will be launched with the job."
+  [{:keys [cpus mem] :as resources}]
+  (if-let [{:keys [cpu-request memory-request]} (-> (config/kubernetes) :sandbox-fileserver :resource-requirements)]
+    (assoc resources
+      :cpus (add-as-decimals cpus cpu-request)
```

Comment:
I think the lowest precision a pod will accept for CPU is `0.001` (i.e., `1m`). If that's true, it might make more sense to just add the doubles and then round to the thousandths place.